File: 3rd-semester/object-oriented-programming/containers/java/src/main/java/com/university/oop/containers/model/BulkContainer.java
package com.university.oop.containers.model;

/**
 * A shipping container carrying bulk (dry) cargo, charged at a flat rate per kilogram.
 *
 * <p>Cost formula: {@code weightKg × 10}</p>
 */
public final class BulkContainer extends Container {

    /** Cost rate per kilogram. */
    private static final double RATE_PER_KG = 10.0;

    private final double weightKg;

    /**
     * Constructs a BulkContainer.
     *
     * @param code        unique container identifier
     * @param destination destination port or city
     * @param weightKg    cargo weight in kilograms
     */
    public BulkContainer(String code, String destination, double weightKg) {
        super(code, destination);
        this.weightKg = weightKg;
    }

    /** @return cargo weight in kilograms */
    public double getWeightKg() { return weightKg; }

    @Override
    public double getCost() { return RATE_PER_KG * weightKg; }

    @Override
    public void printInfo() {
        super.printInfo();
    }

    @Override
    public String toString() {
        return String.format("BulkContainer[code=%s, destination=%s, weight=%.1f kg, cost=%.2f]",
                getCode(), getDestination(), weightKg, getCost());
    }
}

File: 3rd-semester/object-oriented-programming/containers/csharp/src/Models/BulkContainer.cs
namespace ShipCargoSystem.Models;

/// <summary>
/// A shipping container carrying bulk (dry) cargo.
/// <para>Cost formula: <c>weightKg × 10</c></para>
/// </summary>
public sealed class BulkContainer : Container
{
    private const double RatePerKg = 10.0;

    /// <summary>Gets the cargo weight in kilograms.</summary>
    public double WeightKg { get; }

    /// <summary>Initialises a BulkContainer.</summary>
    /// <param name="code">Unique container identifier.</param>
    /// <param name="destination">Destination port or city.</param>
    /// <param name="weightKg">Cargo weight in kilograms.</param>
    public BulkContainer(string code, string destination, double weightKg)
        : base(code, destination) => WeightKg = weightKg;

    /// <inheritdoc/>
    public override double GetCost() => RatePerKg * WeightKg;

    /// <inheritdoc/>
    public override string ToString() =>
        $"BulkContainer[code={Code}, destination={Destination}, weight={WeightKg:F1} kg, cost={GetCost():F2}]";
}
